### PR TITLE
Add version output to all app CI jobs

### PR DESCRIPTION
**Pull request by zarakay**
# Description

I'm working on some docker image build workflows that would allow us to easily have containers built/tagged in a consistent way. In order to have consistent tagging, I want to reuse the version numbers that we use when building releases/snapshots etc. 

This PR adds the version as outputs to all reusable workflows that build apps. Recommend reviewing while hiding whitespace since VScode removed alot of the extra spaces floating around.

# Test Steps

In production lol

# Checklist

If any of these are not applicable, strikethrough the line `~like this~`. **Do not delete it!**. Let the reviewer decide if you should have done it.

### Code
- [x] I have performed a self review of my own code (including checking issues raised when creating the PR).
- [ ] ~I have added/updated unit tests for these changes, and if not I have explained why they are not necessary.~
- [ ] ~I have commented my code in any hard-to-understand or hacky areas.~
- [x] I have handled all new warnings generated by the compiler or IDE.
- [x] I have rebased onto the target branch (usually main).

### Documentation
- [ ] ~I have updated the changelog.~ N/A
- [ ] ~I have updated any documentation required for these changes.~ N/A

# Breaking Changes
- [x] I have considered if this is a breaking change and will communicate it with other team members if so.

### CODE REVIEW

File: .github/workflows/npm-app-snapshot-release.yml [hunk 1]
```diff
@@ -86,11 +91,12 @@ jobs:
         echo "NPM is sensitive to .npmrc formatting; check that it has a newline at the end!!!"
         npm ci --unsafe-perm
         npm run prod
-        # version=$(jq -r .version package.json)
+        version=$(jq -r .version package.json)
```
Comment (zarakay) [inline .github/workflows/npm-app-snapshot-release.yml:94]: Not sure why this was commented out, and what the consequence of putting it back is

File: .github/workflows/csharp-app-release-with-docs.yml [hunk 2]
```diff
@@ -47,16 +47,19 @@ on:
...
         required: false
-      LC_URL: 
+      LC_URL:
         required: false
 
-
+    outputs:
+      version:
+        description: "The current released version."
+        value: ${{ jobs.release-checks.outputs.version }}
```
Comment (alourie) [inline .github/workflows/csharp-app-release-with-docs.yml:62]: nitpick: empty line after the block and before `jobs`

**Review by alourie — COMMENTED** [review]
It looks good, however, can't we  use the released tarballs directly when building containers?

**Comment by zarakay** [comment]
@alourie re: tarball question - From experience at previous jobs where I did containers, I prefer containers that are 'sealed' in how they get built, which means that I can do a `docker build .` and get a working environment without too much trouble. For the last few months I have been manually build EAS for example and its been a pain to ensure that I built the npm, jar, container all correctly. It also makes it easier to have a full docker compose dev env as you can just rebuild the container each time an app changes instead of having to do all the steps separately before building the container.

**Comment by alourie** [comment]
> @alourie re: tarball question - From experience at previous jobs where I did containers, I prefer containers that are 'sealed' in how they get built, which means that I can do a `docker build .` and get a working environment without too much trouble. For the last few months I have been manually build EAS for example and its been a pain to ensure that I built the npm, jar, container all correctly. It also makes it easier to have a full docker compose dev env as you can just rebuild the container each time an app changes instead of having to do all the steps separately before building the container.

Okay, just so I'm clear on the intent - do we plan to use this change to directly build the artefact _and_ the container at one go?

**Comment by zarakay** [comment]
Sort of... Currently my goal is to build only 4 containers... EAS/EWB/Jemena Ingestor/Ewb frontend. I was going to just copy-pasta the workflows for now and see what common-ness appears. I find that docker builds can start out very standard but go in other ways as applications develop. 

TLDR- not right now, but could be one day maybe depending on what sort of weird and wonderful dockerfiles I find along the way. I can already see that EAS+evolve web app will need a custom workflow, there may be others.

**Review by alourie — APPROVED** [review]

File: .github/workflows/npm-app-snapshot-release.yml (hunk 1 above)
Comment (kgreav) [reply .github/workflows/npm-app-snapshot-release.yml:94]: oh god. probs just because it was unused
